Sources/core/objects/Tables.cpp

```diff
@@ -0,0 +1,279 @@
+#include "Tables.h"
+#include "Item.h"
+#include "Human.h"
+
+Tables::Tables(size_t id) : IMovable(id)
+{
+    anchored = true;
+    v_level = 6;
+    up_ = 0;
+    down_ = 0;
+    left_  = 0;
+    right_ = 0; 
+    SetPassable(D_ALL, Passable::SMALL_CREATURE);
+    SetSprite(""); 
+    material_ = "";
+    SetState(material_ + "_table_d0");
+}
+
+void Tables::AfterWorldCreation()
+{
+    CheckSurroundings(true);
+}
+
+void Tables::Delete()
+{
+    CheckSurroundings(false);
+    IOnMapObject::Delete();
+}
+void Tables::CheckSurroundings(bool table)
+{
```

Comment:
Overall, it is better to let table update only itself.
`UpdateSprite(size_t ignored_table)` should scan nearest area for tables (if table id is `ignored_table` then ignore that table) and update only own sprite appropriatly.
Instead `CheckSurroundings` it is better to make some `NotifyNeighborhood(bool is_deleted)` which will scan nearest area for tables and call `UpdateSprite` on them (`UpdateSprite(0)` if it is new table, and `UpdateSprite(GetId())` if table is during deletion).
Overall you have right approach.